test/Microsoft.AspNetCore.Server.KestrelTests/HttpParserTests.cs

```diff
@@ -324,6 +344,44 @@ public void ParseHeadersThrowsOnInvalidRequestHeaders(string rawHeaders, string
             Assert.Equal(StatusCodes.Status400BadRequest, exception.StatusCode);
         }
 
+        [Fact]
+        public void ExceptionDetailNotIncludedWhenLogLevelInformationNotEnabled()
+        {
+            var mockTrace = new Mock<IKestrelTrace>();
+            mockTrace
+                .Setup(trace => trace.IsEnabled(LogLevel.Information))
+                .Returns(false);
+
+            var parser = CreateParser(mockTrace.Object);
+
+            // Invalid request line
+            var buffer = ReadableBuffer.Create(Encoding.ASCII.GetBytes("GET % HTTP/1.1\r\n"));
+
+            var exception = Assert.Throws<BadHttpRequestException>(() =>
+                parser.ParseRequestLine(Mock.Of<IHttpRequestLineHandler>(), buffer, out var consumed, out var examined));
+
+            Assert.Equal("Invalid request line: ''", exception.Message);
```

Comment:
Maybe add a comment here that even though this looks wrong, it's what should happen. The exception isn't used so we didn't waste effort to ToString() the line.